lib/translator/common.h

```diff
@@ -44,13 +46,28 @@ void DeallocPages(void* pages_ptr, uint16_t count);
 void SetAllocPageCallbacks(void* (*alloc_callback)(uint16_t),
                            void (*dealloc_callback)(void*, uint16_t));
 
+struct Allocation {
+  void* data_addr;            // Start of data buffer
+  uint16_t data_page_count;   // Number of data pages
+  void* mdata_addr;           // Start of metadata buffer
+  uint16_t mdata_page_count;  // Number of metadata pages
+};
+
 struct BeginResponse {
   ApiStatus status;
   uint32_t alloc_len;
 };
 
 class Translation {
  public:
+  Translation() {
```

Comment:
perhaps initializer list would be cleaner here
`Translation() : allocations_({nullptr})`

Reply:
The `{nullptr}` here is unnecessary; just `allocations_()` is sufficient to force zero-initialization. See https://en.cppreference.com/w/cpp/language/zero_initialization.